src/components/MovieTile.jsx

```diff
@@ -1,24 +1,67 @@
+import { useRef, useState, useEffect } from "react";
 import PropTypes from 'prop-types';
 import { getReleaseYear } from '../utils/movieUtils';
-import '../components-styles/MovieTile.scss';
+import '../styles/MovieTile.scss';
+import DialogOpenButton from '../components/DialogOpenButton';
 
-const MovieTile = ({poster_path, title, genres, release_date, onClick}) => {
-    const handleClickEvent = (e) => {
-        if (typeof onClick === "function") {
-            onClick(e.currentTarget.value)
+const useOutsideClick = (menuRef, setMenuTileContextOpen) => {
+    useEffect(() => {
+        const handleClickOutside = (event) => {
+            if (menuRef.current && !menuRef.current.contains(event.target)) {
+                setMenuTileContextOpen(false)
+            }
         }
+
+        document.addEventListener("mousedown", handleClickOutside);
+        return () => {
+
+            document.removeEventListener("mousedown", handleClickOutside);
+        };
+    }, [menuRef, setMenuTileContextOpen]);
+}
+
+const MovieTile = ({movieData, onClick}) => {
+    const [isMenuTileContextOpen, setMenuTileContextOpen] = useState(false);// need to hide/show menu
+    const [isMenuTileContextRendered, setMenuTileContextRendered] = useState(false)// need for menu render
```

Comment:
Can we remove it and use setIsMenuOpen?

Reply:
@MykhailoSheliahov no, there is an issue with the context menu. This context menu contains buttons that handle portals. We can have only one to control hidden class.